### ISSUE TITLE
Libraries not being updated to latest versions

### ISSUE THREAD

**Issue by MattIPv4** [cdnjs/cdnjs#14263]
### Discussed in https://github.com/cdnjs/cdnjs/discussions/14262

<div type='discussions-op-text'>

<sup>Originally posted by **heyesr** September 29, 2024</sup>
Hi there,

I've just released a new version of my software - RGraph 6.19 -  but CDNjs doesn't appear to be picking up the new release. The Git Repository is tagged - https://github.com/heyesr/rgraph/tree/6.19 (and branched at this tag too fwiw) but so far nada. Do I just need to be more patient for CDNjs to mirror the file?

Thanks very much!

Richard</div>

**Comment by MattIPv4** [comment]
👀 @xtuc looks like the new version was detected, but wasn't actually processed? https://github.com/cdnjs/logs/tree/prod/packages/R/RGraph/6.19

**Comment by MattIPv4** [comment]
cdnjs does not seem to be updating any libraries, and this appears to have been the case for three or so weeks. I informally escalated this last weekend to no avail, this has now been formally escalated to Cloudflare in #01220256

**Comment by limzykenneth** [comment]
Not sure if it has recovered but the latest release of p5.js (1.11.0) seems to be available now: https://cdnjs.cloudflare.com/ajax/libs/p5.js/1.11.0/p5.min.js. 

It was released about 2 weeks ago and we have noticed at the time it didn't update on CDNjs (stuck on 1.10.0).

**Comment by heyesr** [comment]
Yes, the latest version of RGraph (6.19, https://cdnjs.com/libraries/RGraph/6.19) is available too. Thanks very much for the help!

**Comment by gunjandatta** [comment]
The latest is working for the gd-sprest library too. Thanks for the help.

**Comment by MattIPv4** [comment]
@xtuc put in place a fix to allow new versions to be processed again ❤️ I believe there were a couple of versions that got partially processed as it broke, which will need to be re-processed. Once those are rectified, I'll close out this issue.

**Comment by stevenjoezhang** [comment]
I found that KaTeX has not been updated either (the latest version on npm is 0.16.11, but on cdnjs it is 0.16.9). Will it be fixed? Thanks!

https://www.npmjs.com/package/katex?activeTab=versions
https://cdnjs.com/libraries/KaTeX

**Comment by LeeLenaleee** [comment]
Chart.js version 4.4.6 was released on October the 28. That is 2 weeks after the issue should be fixed but I don't see the version on the CDN. So is there still an issue or is something else the culprit of this?

https://github.com/chartjs/Chart.js/releases/tag/v4.4.6
https://cdnjs.com/libraries/Chart.js
https://www.npmjs.com/package/chart.js?activeTab=versions

**Comment by EvanNotFound** [comment]
Same here for Hexo theme redefine v2.8.2 https://github.com/EvanNotFound/hexo-theme-redefine/tree/v2.8.2

**Comment by echolet** [comment]
[https://cdnjs.com/libraries/quill/2.0.3](https://cdnjs.com/libraries/quill/2.0.3) shows no files

**Comment by gunjandatta** [comment]
> The latest is working for the gd-sprest library too. Thanks for the help.

It stopped working again.

**Comment by lucaswerkmeister** [comment]
https://cdnjs.com/libraries/codex is also missing the latest version (which should be 1.20.1, compare [@wikimedia/codex on npm](https://www.npmjs.com/package/@wikimedia/codex)). The logs repo has a [1.20.0 directory](https://github.com/cdnjs/logs/tree/prod/packages/c/codex/1.20.0), but with fewer files than the [1.19.1 directory](https://github.com/cdnjs/logs/tree/prod/packages/c/codex/1.19.1); meanwhile, the [cdnjs repo](https://github.com/cdnjs/cdnjs/tree/master/ajax/libs/codex/) only goes up to 1.12.0.

**Comment by moappi** [comment]
Also not working for json2html .. latest version should be https://github.com/moappi/json2html/releases/tag/3.3.2 , while cdn is only showing 3.2.2 https://cdnjs.com/libraries/json2html

**Comment by MattIPv4** [comment]
👍 It is not working for any library, this is a known issue the Cloudflare team are working on.

**Comment by alvarotrigo** [comment]
> 👍 It is not working for any library, this is a known issue the Cloudflare team are working on.

That's a pity, time to redirect developers to another CDN so they can get the latest version.

**Comment by MattIPv4** [comment]
Hey folks, not much of a material update, but just to say that I've met with the team working on this again today and things are moving forward to completely redo how cdnjs processes new updates (as well as redoing much of the architecture behind the CDN itself). I don't have a timeline to offer, but it is actively being worked on.

**Comment by UselessGuru** [comment]
https://cdnjs.com/libraries/bootstrap-table is also not getting updated. It is still @ 1.24.0, but latest is 1.24.1:
https://github.com/wenzhixin/bootstrap-table/releases/tag/1.24.1

**Comment by martincostello** [comment]
I see some updates are starting to trickle through 😃 (swagger-ui and plotlyjs)

**Comment by bhaoo** [comment]
Looks like its fixed? 👀

https://github.com/cdnjs/cdnjs/actions/runs/14095081467/job/39480680705

**Comment by MattIPv4** [comment]
Sync runs relate to #14261 -- you'll want to look at cdnjs/logs to see if new library versions are being processed (they are afaik).

**Comment by YamiOdymel** [comment]
Sorry to bother, but I'd like to point out that my package [cdnjs/tocas](https://cdnjs.com/libraries/tocas), which is based on npm, is still listed with an outdated version — it's stuck at v5.0.1 from July 2024.

In contrast, the alternative package [cdnjs/tocas-ui](https://cdnjs.com/libraries/tocas-ui), which pulls from GitHub releases, is fully up to date with v5.0.2 from March 2025.

The comment thread ended with "it seems to be fixed," so I just wanted to make sure this wasn't overlooked.

**Comment by juliangarnier** [comment]
Same here, since the release of Anime.js v4, the cdnjs page properly list v4, but fails loading the files.
https://cdnjs.com/libraries/animejs/4.0.0.
This seems to prevents CodePen from listing the latest version.
Can I do something on my end to solve this?
Any help would be greatly appreciated!
Thanks

**Comment by aavmurphy** [comment]
Bootstrap is not updating either. 2 versions behind.
https://status.cdnjs.com/ does not mention the updating problem, in fact it says 'operational'
It would save folks time if you'd add a 'problems updating versions (and linking here)' message. Repo owners will have spent a lot of time checking its not something they've done.

**Comment by heyesr** [comment]
Hi,

I've just noticed that the latest RGraph libraries are now available on CDNjs - thank you!

Richard

**Comment by aavmurphy** [comment]
Not Bootstrap. Any ETA on a fix? Would it be possible to update Bootstrap manually?

**Comment by MattIPv4** [comment]
I do not have an ETA for a full fix, any updates I do have are communicated here when I have them. It is unfortunately not possible to manually update a library, as it's not just a case of adding it to this git repo -- there are multiple KV buckets that need specific data written to them, an R2 bucket, Algolia indexes, etc.

**Comment by MattIPv4** [comment]
As has been said here repeatedly, this is a known issue affecting many (all) libraries on cdnjs. There is a new team at Cloudflare that is working on rebuilding the auto-update systems as well as overhauling the architecture for cdnjs generally. This is a big project that is going to take them a while, and I do not have an ETA for when it will be completed.

As an update to my previous update, the old auto-update bot that had been started up again briefly has been turned off as far as I'm aware, while the team look at redoing parts of the architecture that the bot interfaced with.

Repeatedly pinging me and posting here is not going to change that, unfortunately. I will post updates here as I have them from the team working on this.

**Comment by MattIPv4** [comment]
I will post updates here as and when I have an update to share.

I help maintain cdnjs in my spare time for free out of the kindness of my heart, so I'm not going to waste my spare time posting updates here every day saying there is no update.

**Comment by heyesr** [comment]
Hi,

Thank you Matt. Cdnjs is a great resource!

Richard
https://www.rgraph.net
Beautiful JavaScript charts for websites

**Comment by BorisMoore** [comment]
This is, to be honest, a real pain. Can you give some kind of status as to how long you expect this issue to persist?

Up to now we had directed our users to the CDNJS urls for CDN delivery. They don't understand why the CDNJS urls for JsRender or JsViews  are no longer working [for the latest version](https://cdnjs.cloudflare.com/ajax/libs/jsrender/1.0.16/jsrender.min.js) - which they need urgently.

I have been forced [to encourage them](https://www.jsviews.com/#download) to switch to using JsDelivr rather than Cdnjs...  

BTW, yes I understand you maintain cdnjs for free, and I appreciate that a  lot. I also am unpaid for the [JsRender/JsViews](https://www.jsviews.com/) work that I do.

**Comment by MattIPv4** [comment]
> Can you give some kind of status as to how long you expect this issue to persist?

I, unfortunately, do not have a timeline to offer, nor have the team at Cloudflare working on overhauling cdnjs offered a timeline. I believe this stems from there being a lot of unknown unknowns as they're slowly discovering all the intracacies (and issues) with the old systems.

**Comment by tribixbite** [comment]
Looks like the update pipeline is back on as of yesterday 🎉 

![Image](https://github.com/user-attachments/assets/b48ddb54-002b-4763-95e4-1fb9cbe31ad9)

**Comment by aavmurphy** [comment]
Good news at last, @tribixbite 

I hesitate to criticise a free resource, but CDNJS / Cloudflare have lost a lot of trust by not warning about this problem on the CDNJS website rather than it being hidden away in a Github issue.

**Comment by MikeKav** [comment]
I was looking at ChartJS, CDNJS is offering 4.4.1 as the latest, which is a year ago.

The github repository shows latest version as 4.5.0 plus a number of releases of 4.4.x, all of which is available on jsdelivr

I'll come back in a couple of months to understand whether version updates are working correctly to understand whether cdnjs is a viable option to return to

**Comment by MattIPv4** [comment]
👋 Hey folks, I've received an update from the Cloudflare folks working on rewriting all the automation here; they're getting close to having the new automation ready to go, which'll handle updating libraries and syncing with this repository. As a final step in that work, they've had to turn off the old automation again for a bit, likely for a few weeks, before the new automation is enabled and running. Apologies for the delays in library updates while this happens.

**Comment by heyesr** [comment]
Thanks for the update!

**Comment by Snrat** [comment]
> 👋 Hey folks, I've received an update from the Cloudflare folks working on rewriting all the automation here; they're getting close to having the new automation ready to go, which'll handle updating libraries and syncing with this repository. As a final step in that work, they've had to turn off the old automation again for a bit, likely for a few weeks, before the new automation is enabled and running. Apologies for the delays in library updates while this happens.

Hi, any update about this?

**Comment by MattIPv4** [comment]
Unfortunately, nothing exciting to report yet; the team is still working on it. You can see test packages being run through the system, though, in https://github.com/cdnjs/logs/, so things are happening!

**Comment by nodesocket** [comment]
Any update? I'd like to see the latest version `tagify` bumped to `v4.36.0` on cdnjs.

**Comment by MattIPv4** [comment]
> Any update?

X-posting my reply from https://github.com/cdnjs/cdnjs/issues/14261#issuecomment-4148164102:

> Unfortunately, it seems that work is taking longer than expected. I meet with the team once a month to check in on things, and from what I've heard, there is still progress being made. I wish I could give further insight, but that's all I have to share, sorry.
> 
> While libraries aren't being updated still, cdnjs itself is operational, which is the main priority always -- websites using cdnjs can continue to load assets without issue, even while migration work is happening in the background to change how those assets are being served.

**Comment by milkyjoe90** [comment]
@MattIPv4, appreciate the update last month, it would be really useful to get even a sense of whether we're a week away, a month, another 6 months?

Could Cloudflare not temporarily turn the old automation back on whilst they are still working on whatever the new system is due to be. Really feels like this is being mothballed as websites that load assets from cdnjs and potentially use it to drive their update cycles won't be seeing potential security updates for their libraries unless they look elsewhere.

**Comment by MattIPv4** [comment]
Unfortunately, I really don't have a great sense of timeline from the team on when the migration will be completed. From what I've understood, they've run into quite a few unexpected issues along the way (I suspect due to cdnjs' scale), and that has caused the timeline to shift. The last I heard was that we're maybe a month or two away, but I really wouldn't bet on that.

I don't think it is viable to have the old system running in parallel from what I've been told, as that would introduce a lot more complexity into the migration work as they'll have libraries changing underneath them while trying to get them all migrated. I imagine it'd only serve to slow things down at this point.

I agree with you, it isn't a great situation, and I've expressed as such to the folks working on this. I wish I had better news, but that is all I have, and I assure you I will keep these issues updated as I hear more.

**Comment by milkyjoe90** [comment]
Thank you Matt, really appreciate the update!

**Comment by milkyjoe90** [comment]
@MattIPv4 Can we prod you again for an update as you'll hopefully have met with Cloudflare at least once since your last comment on here?

**Comment by MattIPv4** [comment]
Unfortunately, no update to give here on getting libraries updated. Work is still progressing, and the team recently completed the cutover for how existing libraries are served on the CDN. As I hear more relating to actually getting things moving again, I will make sure to update here.

**Comment by Snrat** [comment]
> Unfortunately, no update to give here on getting libraries updated. Work is still progressing, and the team recently completed the cutover for how existing libraries are served on the CDN. As I hear more relating to actually getting things moving again, I will make sure to update here.

Thank you for the update. As others have mentioned, this repo has not been synced for more than nine months, and it may still remain unsynced by the time it reaches one year.

As a result, there are growing concerns about the safety and reliability of referencing these components. This also means that many users can no longer continue using cdnjs and may have to switch to jsDelivr or similar services instead.

Could you please let us know whether the team currently has any plans or timeline for addressing this?

**Comment by MattIPv4** [comment]
> Thank you for the update. As others have mentioned, this repo has not been synced for more than nine months, and it may still remain unsynced by the time it reaches one year.

For what it is worth, I suspect this repository may never be synced again. It is not the source of truth for the CDN and has not been for a long time now -- we've merely maintained it as a mirror of the CDN itself to allow folks to clone it as a way to operate their own mirrors. If we do move forward with deprecating this repository, we will provide a new means of maintaining mirrors.

> Could you please let us know whether the team currently has any plans or timeline for addressing this?

I believe I have explained this adnauseum above already -- I do not have a timeline I can share, if I have one I will share it.

> As a result, there are growing concerns about the safety and reliability of referencing these components. 

Reliability of the CDN itself is always the priority. We are very careful to ensure that existing assets, relied on by ~12% of the entire internet, continue to be served without issue. If you encounter any issues with the CDN itself, please let us know.

**Comment by MattIPv4** [comment]
Hey folks, good news! We've got the new auto-update automation up and running, and you can see it starting to work through the backlog of library versions over in https://github.com/cdnjs/logs

Obviously, it is going to take quite a while for it to get through everything that has been missed over the last year or so, and there's a good chance we might need to pause it if/when we encounter any teething issues. I asked that you have some patience as it does its thing. Please do shout if you see any issues with the versions that it has processed!

**Comment by nodesocket** [comment]
@MattIPv4 awesome, thanks for staying on top of this and getting over the finish line 🏁 . Appreciated 🙏🏻

**Comment by MattIPv4** [comment]
No worries. I should be clear that much of this is very much a royal we, I'm just the one providing updates here, the folks at Cloudflare are the ones that've put in all the hard work to rewrite the automation. I'm sure there'll be more info shared by them in due course about this.

**Comment by heyesr** [comment]
Congrats and thanks for the update! :-)

**Comment by milkyjoe90** [comment]
This is great, thank you so, so much for the update Matt. So this should mean that once the automation has caught up, the API for cdnjs etc will start reporting out the latest versions? For my purposes getting libraries, versions and hashes from the API is more than sufficient, the repo itself being synced is less of a concern for me.

Please pass on mine and others thanks to the folks at Cloudflare for making progress on this. It's massively appreciated.

**Comment by MattIPv4** [comment]
> So this should mean that once the automation has caught up, the API for cdnjs etc will start reporting out the latest versions?

Indeed, as the bot starts processing new versions they'll show up on the API.

**Comment by mirabilos** [comment]
same happens for me (dygraphs) every time I release it, unpkg picks it up almost immediately

**Comment by MattIPv4** [comment]
> same happens for me (dygraphs) every time I release it, unpkg picks it up almost immediately

https://github.com/cdnjs/logs/tree/prod/packages/d/dygraph

https://github.com/cdnjs/packages/blob/master/packages/d/dygraph.json

https://www.npmjs.com/package/dygraphs?activeTab=versions

All your previous versions released many years ago have been picked up by cdnjs just fine -- looks like we're just missing the most recent version published a day ago. I suspect the bot is still catching up from when it was re-enabled, but if we don't pick up the new version in the next few days, let me know and we can dig further.

**Comment by mirabilos** [comment]
On Tue, 28 Jul 2026, Matt Cowley wrote:

>MattIPv4 left a comment (cdnjs/cdnjs#14263)

(why not IPv6?)

>All your previous versions released many years ago have been picked up
>by cdnjs just fine -- looks like we're just missing the most recent
>version published a day ago. I suspect the bot is still catching up
>from when it was re-enabled, but if we don't pick up the new version in
>the next few days, let me know and we can dig further.

Yeah. Thanks!

**Comment by japui-coder** [comment]
Libraries not being updated to latest versions
 #14263

**Comment by vndroid** [comment]
`highlight.js` seems that there has been no update.

**Comment by MattIPv4** [comment]
@vndroid `highlight.js` was last updated a month ago for the `11.11.2` release: https://github.com/cdnjs/logs/tree/prod/packages/h/highlight.js

Looking at the config for the package (https://github.com/cdnjs/packages/blob/master/packages/h/highlight.js.json), it is set to pull from https://www.npmjs.com/package/@highlightjs/cdn-assets, which does have a newer `11.12.0` release that we haven't picked up even though the file map appears to still match.

@simonabadoiu can you take a look at why the bot isn't picking up that version, given it has been published for a month? Is there any further diagnosis I can do from the outside?

**Comment by vndroid** [comment]
> [@vndroid](https://github.com/vndroid) `highlight.js` was last updated a month ago for the `11.11.2` release: https://github.com/cdnjs/logs/tree/prod/packages/h/highlight.js
> 
> Looking at the config for the package (https://github.com/cdnjs/packages/blob/master/packages/h/highlight.js.json), it is set to pull from https://www.npmjs.com/package/@highlightjs/cdn-assets, which does have a newer `11.12.0` release that we haven't picked up even though the file map appears to still match.
> 
> [@simonabadoiu](https://github.com/simonabadoiu) can you take a look at why the bot isn't picking up that version, given it has been published for a month? Is there any further diagnosis I can do from the outside?

By the way, the [mathjax](https://cdnjs.com/libraries/mathjax/4.1.3) , [font-awesome](https://cdnjs.com/libraries/font-awesome/7.3.1) and [tocbot](https://cdnjs.com/libraries/tocbot/4.36.8) could not be found.
